pkg/fs/dir_others.go

```diff
@@ -0,0 +1,70 @@
+// +build windows solaris
```

Comment:
We can add `!linux !darwin !rest of the bsd's ` - that way this code works for all other platforms if any.